ponce <matteo.bernardini@gmail.com>

Qualify placeholders with their full namespace.

This is needed with boost >= 1.73.0

diff -Naur gnuradio-3.7.13.2.orig/docs/doxygen/other/msg_passing.dox gnuradio-3.7.13.2/docs/doxygen/other/msg_passing.dox
--- gnuradio-3.7.13.2.orig/docs/doxygen/other/msg_passing.dox	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/docs/doxygen/other/msg_passing.dox	2020-05-17 10:25:08.170538000 +0200
@@ -101,7 +101,7 @@
 
 \code
   set_msg_handler(pmt::pmt_t port_id,
-    boost::bind(&block_class::message_handler_function, this, _1));
+    boost::bind(&block_class::message_handler_function, this, boost::placeholders::_1));
 \endcode
 
 In Python:
@@ -276,15 +276,15 @@
 {
   message_port_register_in(pmt::mp("print"));
   set_msg_handler(pmt::mp("print"),
-    boost::bind(&message_debug_impl::print, this, _1));
+    boost::bind(&message_debug_impl::print, this, boost::placeholders::_1));
 
   message_port_register_in(pmt::mp("store"));
   set_msg_handler(pmt::mp("store"),
-    boost::bind(&message_debug_impl::store, this, _1));
+    boost::bind(&message_debug_impl::store, this, boost::placeholders::_1));
 
   message_port_register_in(pmt::mp("print_pdu"));
   set_msg_handler(pmt::mp("print_pdu"),
-    boost::bind(&message_debug_impl::print_pdu, this, _1));
+    boost::bind(&message_debug_impl::print_pdu, this, boost::placeholders::_1));
 }
 \endcode
 
diff -Naur gnuradio-3.7.13.2.orig/gnuradio-runtime/lib/block.cc gnuradio-3.7.13.2/gnuradio-runtime/lib/block.cc
--- gnuradio-3.7.13.2.orig/gnuradio-runtime/lib/block.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gnuradio-runtime/lib/block.cc	2020-05-17 10:25:08.164538000 +0200
@@ -60,7 +60,7 @@
   {
     global_block_registry.register_primitive(alias(), this);
     message_port_register_in(d_system_port);
-    set_msg_handler(d_system_port, boost::bind(&block::system_handler, this, _1));
+    set_msg_handler(d_system_port, boost::bind(&block::system_handler, this, boost::placeholders::_1));
 
     configure_default_loggers(d_logger, d_debug_logger, symbol_name());
   }
diff -Naur gnuradio-3.7.13.2.orig/gr-analog/lib/sig_source_X_impl.cc.t gnuradio-3.7.13.2/gr-analog/lib/sig_source_X_impl.cc.t
--- gnuradio-3.7.13.2.orig/gr-analog/lib/sig_source_X_impl.cc.t	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-analog/lib/sig_source_X_impl.cc.t	2020-05-17 10:25:08.169538000 +0200
@@ -55,7 +55,7 @@
       set_frequency(frequency);
 
       message_port_register_in(pmt::mp("freq"));
-      set_msg_handler(pmt::mp("freq"), boost::bind(&@IMPL_NAME@::set_frequency_msg, this, _1));
+      set_msg_handler(pmt::mp("freq"), boost::bind(&@IMPL_NAME@::set_frequency_msg, this, boost::placeholders::_1));
     }
 
     @IMPL_NAME@::~@IMPL_NAME@()
diff -Naur gnuradio-3.7.13.2.orig/gr-blocks/lib/copy_impl.cc gnuradio-3.7.13.2/gr-blocks/lib/copy_impl.cc
--- gnuradio-3.7.13.2.orig/gr-blocks/lib/copy_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-blocks/lib/copy_impl.cc	2020-05-17 10:25:08.162538000 +0200
@@ -47,7 +47,7 @@
     {
       message_port_register_in(pmt::mp("en"));
       set_msg_handler(pmt::mp("en"),
-                      boost::bind(&copy_impl::handle_enable, this, _1));
+                      boost::bind(&copy_impl::handle_enable, this, boost::placeholders::_1));
     }
 
     copy_impl::~copy_impl()
diff -Naur gnuradio-3.7.13.2.orig/gr-blocks/lib/message_debug_impl.cc gnuradio-3.7.13.2/gr-blocks/lib/message_debug_impl.cc
--- gnuradio-3.7.13.2.orig/gr-blocks/lib/message_debug_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-blocks/lib/message_debug_impl.cc	2020-05-17 10:25:08.160538000 +0200
@@ -102,13 +102,13 @@
                  io_signature::make(0, 0, 0))
     {
       message_port_register_in(pmt::mp("print"));
-      set_msg_handler(pmt::mp("print"), boost::bind(&message_debug_impl::print, this, _1));
+      set_msg_handler(pmt::mp("print"), boost::bind(&message_debug_impl::print, this, boost::placeholders::_1));
 
       message_port_register_in(pmt::mp("store"));
-      set_msg_handler(pmt::mp("store"), boost::bind(&message_debug_impl::store, this, _1));
+      set_msg_handler(pmt::mp("store"), boost::bind(&message_debug_impl::store, this, boost::placeholders::_1));
 
       message_port_register_in(pmt::mp("print_pdu"));
-      set_msg_handler(pmt::mp("print_pdu"), boost::bind(&message_debug_impl::print_pdu, this, _1));
+      set_msg_handler(pmt::mp("print_pdu"), boost::bind(&message_debug_impl::print_pdu, this, boost::placeholders::_1));
     }
 
     message_debug_impl::~message_debug_impl()
diff -Naur gnuradio-3.7.13.2.orig/gr-blocks/lib/message_strobe_impl.cc gnuradio-3.7.13.2/gr-blocks/lib/message_strobe_impl.cc
--- gnuradio-3.7.13.2.orig/gr-blocks/lib/message_strobe_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-blocks/lib/message_strobe_impl.cc	2020-05-17 10:25:08.159538000 +0200
@@ -58,7 +58,7 @@
 
       message_port_register_in(pmt::mp("set_msg"));
       set_msg_handler(pmt::mp("set_msg"),
-                      boost::bind(&message_strobe_impl::set_msg, this, _1));
+                      boost::bind(&message_strobe_impl::set_msg, this, boost::placeholders::_1));
     }
 
     message_strobe_impl::~message_strobe_impl()
diff -Naur gnuradio-3.7.13.2.orig/gr-blocks/lib/message_strobe_random_impl.cc gnuradio-3.7.13.2/gr-blocks/lib/message_strobe_random_impl.cc
--- gnuradio-3.7.13.2.orig/gr-blocks/lib/message_strobe_random_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-blocks/lib/message_strobe_random_impl.cc	2020-05-17 10:25:08.163538000 +0200
@@ -68,7 +68,7 @@
 
       message_port_register_in(pmt::mp("set_msg"));
       set_msg_handler(pmt::mp("set_msg"),
-                      boost::bind(&message_strobe_random_impl::set_msg, this, _1));
+                      boost::bind(&message_strobe_random_impl::set_msg, this, boost::placeholders::_1));
     }
 
     float message_strobe_random_impl::next_delay(){
diff -Naur gnuradio-3.7.13.2.orig/gr-blocks/lib/multiply_matrix_cc_impl.cc gnuradio-3.7.13.2/gr-blocks/lib/multiply_matrix_cc_impl.cc
--- gnuradio-3.7.13.2.orig/gr-blocks/lib/multiply_matrix_cc_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-blocks/lib/multiply_matrix_cc_impl.cc	2020-05-17 10:25:08.161538000 +0200
@@ -57,7 +57,7 @@
       message_port_register_in(port_name);
       set_msg_handler(
           port_name,
-          boost::bind(&multiply_matrix_cc_impl::msg_handler_A, this, _1)
+          boost::bind(&multiply_matrix_cc_impl::msg_handler_A, this, boost::placeholders::_1)
       );
     }
 
diff -Naur gnuradio-3.7.13.2.orig/gr-blocks/lib/multiply_matrix_ff_impl.cc gnuradio-3.7.13.2/gr-blocks/lib/multiply_matrix_ff_impl.cc
--- gnuradio-3.7.13.2.orig/gr-blocks/lib/multiply_matrix_ff_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-blocks/lib/multiply_matrix_ff_impl.cc	2020-05-17 10:25:08.162538000 +0200
@@ -57,7 +57,7 @@
       message_port_register_in(port_name);
       set_msg_handler(
           port_name,
-          boost::bind(&multiply_matrix_ff_impl::msg_handler_A, this, _1)
+          boost::bind(&multiply_matrix_ff_impl::msg_handler_A, this, boost::placeholders::_1)
       );
     }
 
diff -Naur gnuradio-3.7.13.2.orig/gr-blocks/lib/mute_XX_impl.cc.t gnuradio-3.7.13.2/gr-blocks/lib/mute_XX_impl.cc.t
--- gnuradio-3.7.13.2.orig/gr-blocks/lib/mute_XX_impl.cc.t	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-blocks/lib/mute_XX_impl.cc.t	2020-05-17 10:25:08.163538000 +0200
@@ -48,7 +48,7 @@
     {
       message_port_register_in(pmt::intern("set_mute"));
       set_msg_handler(pmt::intern("set_mute"),
-        boost::bind(&@NAME_IMPL@::set_mute_pmt, this, _1));
+        boost::bind(&@NAME_IMPL@::set_mute_pmt, this, boost::placeholders::_1));
     }
 
     @NAME_IMPL@::~@NAME_IMPL@()
diff -Naur gnuradio-3.7.13.2.orig/gr-blocks/lib/nop_impl.cc gnuradio-3.7.13.2/gr-blocks/lib/nop_impl.cc
--- gnuradio-3.7.13.2.orig/gr-blocks/lib/nop_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-blocks/lib/nop_impl.cc	2020-05-17 10:25:08.164538000 +0200
@@ -46,7 +46,7 @@
     {
       // Arrange to have count_received_msgs called when messages are received.
       message_port_register_in(pmt::mp("port"));
-      set_msg_handler(pmt::mp("port"), boost::bind(&nop_impl::count_received_msgs, this, _1));
+      set_msg_handler(pmt::mp("port"), boost::bind(&nop_impl::count_received_msgs, this, boost::placeholders::_1));
     }
 
     nop_impl::~nop_impl()
diff -Naur gnuradio-3.7.13.2.orig/gr-blocks/lib/pdu_filter_impl.cc gnuradio-3.7.13.2/gr-blocks/lib/pdu_filter_impl.cc
--- gnuradio-3.7.13.2.orig/gr-blocks/lib/pdu_filter_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-blocks/lib/pdu_filter_impl.cc	2020-05-17 10:25:08.158538000 +0200
@@ -45,7 +45,7 @@
     {
       message_port_register_out(pdu::pdu_port_id());
       message_port_register_in(pdu::pdu_port_id());
-      set_msg_handler(pdu::pdu_port_id(), boost::bind(&pdu_filter_impl::handle_msg, this, _1));
+      set_msg_handler(pdu::pdu_port_id(), boost::bind(&pdu_filter_impl::handle_msg, this, boost::placeholders::_1));
     }
 
     void
diff -Naur gnuradio-3.7.13.2.orig/gr-blocks/lib/pdu_remove_impl.cc gnuradio-3.7.13.2/gr-blocks/lib/pdu_remove_impl.cc
--- gnuradio-3.7.13.2.orig/gr-blocks/lib/pdu_remove_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-blocks/lib/pdu_remove_impl.cc	2020-05-17 10:25:08.163538000 +0200
@@ -45,7 +45,7 @@
     {
       message_port_register_out(pdu::pdu_port_id());
       message_port_register_in(pdu::pdu_port_id());
-      set_msg_handler(pdu::pdu_port_id(), boost::bind(&pdu_remove_impl::handle_msg, this, _1));
+      set_msg_handler(pdu::pdu_port_id(), boost::bind(&pdu_remove_impl::handle_msg, this, boost::placeholders::_1));
     }
 
     void
diff -Naur gnuradio-3.7.13.2.orig/gr-blocks/lib/pdu_set_impl.cc gnuradio-3.7.13.2/gr-blocks/lib/pdu_set_impl.cc
--- gnuradio-3.7.13.2.orig/gr-blocks/lib/pdu_set_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-blocks/lib/pdu_set_impl.cc	2020-05-17 10:25:08.159538000 +0200
@@ -45,7 +45,7 @@
     {
       message_port_register_out(pdu::pdu_port_id());
       message_port_register_in(pdu::pdu_port_id());
-      set_msg_handler(pdu::pdu_port_id(), boost::bind(&pdu_set_impl::handle_msg, this, _1));
+      set_msg_handler(pdu::pdu_port_id(), boost::bind(&pdu_set_impl::handle_msg, this, boost::placeholders::_1));
     }
 
     void
diff -Naur gnuradio-3.7.13.2.orig/gr-blocks/lib/random_pdu_impl.cc gnuradio-3.7.13.2/gr-blocks/lib/random_pdu_impl.cc
--- gnuradio-3.7.13.2.orig/gr-blocks/lib/random_pdu_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-blocks/lib/random_pdu_impl.cc	2020-05-17 10:25:08.162538000 +0200
@@ -50,7 +50,7 @@
     {
       message_port_register_out(pdu::pdu_port_id());
       message_port_register_in(pmt::mp("generate"));
-      set_msg_handler(pmt::mp("generate"), boost::bind(&random_pdu_impl::generate_pdu, this, _1));
+      set_msg_handler(pmt::mp("generate"), boost::bind(&random_pdu_impl::generate_pdu, this, boost::placeholders::_1));
       if(length_modulo < 1)
         throw std::runtime_error("length_module must be >= 1");
       if(max_items < length_modulo)
diff -Naur gnuradio-3.7.13.2.orig/gr-blocks/lib/repeat_impl.cc gnuradio-3.7.13.2/gr-blocks/lib/repeat_impl.cc
--- gnuradio-3.7.13.2.orig/gr-blocks/lib/repeat_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-blocks/lib/repeat_impl.cc	2020-05-17 10:25:08.160538000 +0200
@@ -45,7 +45,7 @@
     {
         message_port_register_in(pmt::mp("interpolation"));
         set_msg_handler(pmt::mp("interpolation"),
-                boost::bind(&repeat_impl::msg_set_interpolation, this, _1));
+                boost::bind(&repeat_impl::msg_set_interpolation, this, boost::placeholders::_1));
     }
 
     void
diff -Naur gnuradio-3.7.13.2.orig/gr-blocks/lib/socket_pdu_impl.cc gnuradio-3.7.13.2/gr-blocks/lib/socket_pdu_impl.cc
--- gnuradio-3.7.13.2.orig/gr-blocks/lib/socket_pdu_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-blocks/lib/socket_pdu_impl.cc	2020-05-17 10:25:08.161538000 +0200
@@ -86,7 +86,7 @@
 
         start_tcp_accept();
 
-        set_msg_handler(pdu::pdu_port_id(), boost::bind(&socket_pdu_impl::tcp_server_send, this, _1));
+        set_msg_handler(pdu::pdu_port_id(), boost::bind(&socket_pdu_impl::tcp_server_send, this, boost::placeholders::_1));
       }
       else if (type =="TCP_CLIENT") {
         boost::system::error_code error = boost::asio::error::host_not_found;
@@ -96,7 +96,7 @@
             throw boost::system::system_error(error);
         d_tcp_socket->set_option(boost::asio::ip::tcp::no_delay(d_tcp_no_delay));
 
-        set_msg_handler(pdu::pdu_port_id(), boost::bind(&socket_pdu_impl::tcp_client_send, this, _1));
+        set_msg_handler(pdu::pdu_port_id(), boost::bind(&socket_pdu_impl::tcp_client_send, this, boost::placeholders::_1));
 
         d_tcp_socket->async_read_some(boost::asio::buffer(d_rxbuf),
           boost::bind(&socket_pdu_impl::handle_tcp_read, this,
@@ -110,7 +110,7 @@
             boost::asio::placeholders::error,
             boost::asio::placeholders::bytes_transferred));
 
-        set_msg_handler(pdu::pdu_port_id(), boost::bind(&socket_pdu_impl::udp_send, this, _1));
+        set_msg_handler(pdu::pdu_port_id(), boost::bind(&socket_pdu_impl::udp_send, this, boost::placeholders::_1));
       }
       else if (type =="UDP_CLIENT") {
         d_udp_socket.reset(new boost::asio::ip::udp::socket(d_io_service, d_udp_endpoint));
@@ -119,7 +119,7 @@
             boost::asio::placeholders::error,
             boost::asio::placeholders::bytes_transferred));
 
-        set_msg_handler(pdu::pdu_port_id(), boost::bind(&socket_pdu_impl::udp_send, this, _1));
+        set_msg_handler(pdu::pdu_port_id(), boost::bind(&socket_pdu_impl::udp_send, this, boost::placeholders::_1));
       }
       else
         throw std::runtime_error("gr::blocks:socket_pdu: unknown socket type");
diff -Naur gnuradio-3.7.13.2.orig/gr-blocks/lib/tagged_stream_multiply_length_impl.cc gnuradio-3.7.13.2/gr-blocks/lib/tagged_stream_multiply_length_impl.cc
--- gnuradio-3.7.13.2.orig/gr-blocks/lib/tagged_stream_multiply_length_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-blocks/lib/tagged_stream_multiply_length_impl.cc	2020-05-17 10:25:08.163538000 +0200
@@ -48,7 +48,7 @@
       set_relative_rate(1);
       message_port_register_in(pmt::intern("set_scalar"));
       set_msg_handler(pmt::intern("set_scalar"),
-        boost::bind(&tagged_stream_multiply_length_impl::set_scalar_pmt, this, _1));
+        boost::bind(&tagged_stream_multiply_length_impl::set_scalar_pmt, this, boost::placeholders::_1));
     }
 
     tagged_stream_multiply_length_impl::~tagged_stream_multiply_length_impl()
diff -Naur gnuradio-3.7.13.2.orig/gr-blocks/lib/tuntap_pdu_impl.cc gnuradio-3.7.13.2/gr-blocks/lib/tuntap_pdu_impl.cc
--- gnuradio-3.7.13.2.orig/gr-blocks/lib/tuntap_pdu_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-blocks/lib/tuntap_pdu_impl.cc	2020-05-17 10:25:08.163538000 +0200
@@ -97,7 +97,7 @@
 
       // set up input message port
       message_port_register_in(pdu::pdu_port_id());
-      set_msg_handler(pdu::pdu_port_id(), boost::bind(&tuntap_pdu_impl::send, this, _1));
+      set_msg_handler(pdu::pdu_port_id(), boost::bind(&tuntap_pdu_impl::send, this, boost::placeholders::_1));
     }
 
     int
diff -Naur gnuradio-3.7.13.2.orig/gr-digital/lib/chunks_to_symbols_XX_impl.cc.t gnuradio-3.7.13.2/gr-digital/lib/chunks_to_symbols_XX_impl.cc.t
--- gnuradio-3.7.13.2.orig/gr-digital/lib/chunks_to_symbols_XX_impl.cc.t	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-digital/lib/chunks_to_symbols_XX_impl.cc.t	2020-05-17 10:31:33.972538000 +0200
@@ -52,7 +52,7 @@
       set_msg_handler(
         pmt::mp("set_symbol_table"),
         boost::bind(&@IMPL_NAME@::handle_set_symbol_table,
-                    this, _1));
+                    this, boost::placeholders::_1));
     }
 
     @IMPL_NAME@::~@IMPL_NAME@()
diff -Naur gnuradio-3.7.13.2.orig/gr-digital/lib/constellation_receiver_cb_impl.cc gnuradio-3.7.13.2/gr-digital/lib/constellation_receiver_cb_impl.cc
--- gnuradio-3.7.13.2.orig/gr-digital/lib/constellation_receiver_cb_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-digital/lib/constellation_receiver_cb_impl.cc	2020-05-17 10:30:11.359538000 +0200
@@ -65,12 +65,12 @@
       set_msg_handler(
         pmt::mp("set_constellation"),
         boost::bind(&constellation_receiver_cb_impl::handle_set_constellation,
-                    this, _1));
+                    this, boost::placeholders::_1));
 
       message_port_register_in(pmt::mp("rotate_phase"));
       set_msg_handler(pmt::mp("rotate_phase"),
                       boost::bind(&constellation_receiver_cb_impl::handle_rotate_phase,
-                                  this, _1));
+                                  this, boost::placeholders::_1));
     }
 
     constellation_receiver_cb_impl::~constellation_receiver_cb_impl()
diff -Naur gnuradio-3.7.13.2.orig/gr-digital/lib/costas_loop_cc_impl.cc gnuradio-3.7.13.2/gr-digital/lib/costas_loop_cc_impl.cc
--- gnuradio-3.7.13.2.orig/gr-digital/lib/costas_loop_cc_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-digital/lib/costas_loop_cc_impl.cc	2020-05-17 10:30:55.808538000 +0200
@@ -83,7 +83,7 @@
       set_msg_handler(
         pmt::mp("noise"),
         boost::bind(&costas_loop_cc_impl::handle_set_noise,
-                    this, _1));
+                    this, boost::placeholders::_1));
     }
 
     costas_loop_cc_impl::~costas_loop_cc_impl()
diff -Naur gnuradio-3.7.13.2.orig/gr-digital/lib/crc32_async_bb_impl.cc gnuradio-3.7.13.2/gr-digital/lib/crc32_async_bb_impl.cc
--- gnuradio-3.7.13.2.orig/gr-digital/lib/crc32_async_bb_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-digital/lib/crc32_async_bb_impl.cc	2020-05-17 10:25:08.172538000 +0200
@@ -51,9 +51,9 @@
       message_port_register_out(d_out_port);
 
       if(check)
-        set_msg_handler(d_in_port, boost::bind(&crc32_async_bb_impl::check, this ,_1) );
+        set_msg_handler(d_in_port, boost::bind(&crc32_async_bb_impl::check, this , boost::placeholders::_1) );
       else
-        set_msg_handler(d_in_port, boost::bind(&crc32_async_bb_impl::calc, this ,_1) );
+        set_msg_handler(d_in_port, boost::bind(&crc32_async_bb_impl::calc, this , boost::placeholders::_1) );
     }
 
     crc32_async_bb_impl::~crc32_async_bb_impl()
diff -Naur gnuradio-3.7.13.2.orig/gr-digital/lib/header_payload_demux_impl.cc gnuradio-3.7.13.2/gr-digital/lib/header_payload_demux_impl.cc
--- gnuradio-3.7.13.2.orig/gr-digital/lib/header_payload_demux_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-digital/lib/header_payload_demux_impl.cc	2020-05-17 10:25:08.172538000 +0200
@@ -154,7 +154,7 @@
       }
       set_tag_propagation_policy(TPP_DONT);
       message_port_register_in(msg_port_id());
-      set_msg_handler(msg_port_id(), boost::bind(&header_payload_demux_impl::parse_header_data_msg, this, _1));
+      set_msg_handler(msg_port_id(), boost::bind(&header_payload_demux_impl::parse_header_data_msg, this, boost::placeholders::_1));
       for (size_t i = 0; i < special_tags.size(); i++) {
         d_special_tags.push_back(pmt::string_to_symbol(special_tags[i]));
         d_special_tags_last_value.push_back(pmt::PMT_NIL);
diff -Naur gnuradio-3.7.13.2.orig/gr-digital/lib/protocol_formatter_async_impl.cc gnuradio-3.7.13.2/gr-digital/lib/protocol_formatter_async_impl.cc
--- gnuradio-3.7.13.2.orig/gr-digital/lib/protocol_formatter_async_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-digital/lib/protocol_formatter_async_impl.cc	2020-05-17 10:25:08.172538000 +0200
@@ -55,7 +55,7 @@
       message_port_register_out(d_pld_port);
 
       set_msg_handler(d_in_port,
-                      boost::bind(&protocol_formatter_async_impl::append, this ,_1) );
+                      boost::bind(&protocol_formatter_async_impl::append, this , boost::placeholders::_1) );
     }
 
     protocol_formatter_async_impl::~protocol_formatter_async_impl()
diff -Naur gnuradio-3.7.13.2.orig/gr-fec/lib/async_decoder_impl.cc gnuradio-3.7.13.2/gr-fec/lib/async_decoder_impl.cc
--- gnuradio-3.7.13.2.orig/gr-fec/lib/async_decoder_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-fec/lib/async_decoder_impl.cc	2020-05-17 10:25:08.170538000 +0200
@@ -67,10 +67,10 @@
 
       if(d_packed) {
         d_pack = new blocks::kernel::pack_k_bits(8);
-        set_msg_handler(d_in_port, boost::bind(&async_decoder_impl::decode_packed, this ,_1));
+        set_msg_handler(d_in_port, boost::bind(&async_decoder_impl::decode_packed, this , boost::placeholders::_1));
       }
       else {
-        set_msg_handler(d_in_port, boost::bind(&async_decoder_impl::decode_unpacked, this ,_1));
+        set_msg_handler(d_in_port, boost::bind(&async_decoder_impl::decode_unpacked, this , boost::placeholders::_1));
       }
 
       // The maximum frame size is set by the initial frame size of the decoder.
diff -Naur gnuradio-3.7.13.2.orig/gr-fec/lib/async_encoder_impl.cc gnuradio-3.7.13.2/gr-fec/lib/async_encoder_impl.cc
--- gnuradio-3.7.13.2.orig/gr-fec/lib/async_encoder_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-fec/lib/async_encoder_impl.cc	2020-05-17 10:25:08.169538000 +0200
@@ -63,7 +63,7 @@
       message_port_register_out(d_out_port);
 
       if(d_packed) {
-        set_msg_handler(d_in_port, boost::bind(&async_encoder_impl::encode_packed, this ,_1) );
+        set_msg_handler(d_in_port, boost::bind(&async_encoder_impl::encode_packed, this , boost::placeholders::_1) );
 
         d_unpack = new blocks::kernel::unpack_k_bits(8);
 
@@ -73,7 +73,7 @@
 
       }
       else {
-        set_msg_handler(d_in_port, boost::bind(&async_encoder_impl::encode_unpacked, this ,_1) );
+        set_msg_handler(d_in_port, boost::bind(&async_encoder_impl::encode_unpacked, this , boost::placeholders::_1) );
       }
 
       if(d_packed || (strncmp(d_encoder->get_input_conversion(), "pack", 4) == 0)) {
diff -Naur gnuradio-3.7.13.2.orig/gr-fec/lib/depuncture_bb_impl.cc gnuradio-3.7.13.2/gr-fec/lib/depuncture_bb_impl.cc
--- gnuradio-3.7.13.2.orig/gr-fec/lib/depuncture_bb_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-fec/lib/depuncture_bb_impl.cc	2020-05-17 10:25:08.170538000 +0200
@@ -76,7 +76,7 @@
       set_fixed_rate(true);
       set_relative_rate((double)d_puncsize/(d_puncsize - d_puncholes));
       set_output_multiple(d_puncsize);
-      //set_msg_handler(boost::bind(&depuncture_bb_impl::catch_msg, this, _1));
+      //set_msg_handler(boost::bind(&depuncture_bb_impl::catch_msg, this, boost::placeholders::_1));
     }
 
     depuncture_bb_impl::~depuncture_bb_impl()
diff -Naur gnuradio-3.7.13.2.orig/gr-fec/lib/puncture_bb_impl.cc gnuradio-3.7.13.2/gr-fec/lib/puncture_bb_impl.cc
--- gnuradio-3.7.13.2.orig/gr-fec/lib/puncture_bb_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-fec/lib/puncture_bb_impl.cc	2020-05-17 10:25:08.170538000 +0200
@@ -73,7 +73,7 @@
       set_fixed_rate(true);
       set_relative_rate((double)(d_puncsize - d_puncholes)/d_puncsize);
       set_output_multiple(d_puncsize - d_puncholes);
-      //set_msg_handler(boost::bind(&puncture_bb_impl::catch_msg, this, _1));
+      //set_msg_handler(boost::bind(&puncture_bb_impl::catch_msg, this, boost::placeholders::_1));
     }
 
     puncture_bb_impl::~puncture_bb_impl()
diff -Naur gnuradio-3.7.13.2.orig/gr-fec/lib/puncture_ff_impl.cc gnuradio-3.7.13.2/gr-fec/lib/puncture_ff_impl.cc
--- gnuradio-3.7.13.2.orig/gr-fec/lib/puncture_ff_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-fec/lib/puncture_ff_impl.cc	2020-05-17 10:25:08.170538000 +0200
@@ -73,7 +73,7 @@
       set_fixed_rate(true);
       set_relative_rate((double)(d_puncsize - d_puncholes)/d_puncsize);
       set_output_multiple(d_puncsize - d_puncholes);
-      //set_msg_handler(boost::bind(&puncture_ff_impl::catch_msg, this, _1));
+      //set_msg_handler(boost::bind(&puncture_ff_impl::catch_msg, this, boost::placeholders::_1));
     }
 
     puncture_ff_impl::~puncture_ff_impl()
diff -Naur gnuradio-3.7.13.2.orig/gr-filter/lib/fractional_resampler_cc_impl.cc gnuradio-3.7.13.2/gr-filter/lib/fractional_resampler_cc_impl.cc
--- gnuradio-3.7.13.2.orig/gr-filter/lib/fractional_resampler_cc_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-filter/lib/fractional_resampler_cc_impl.cc	2020-05-17 10:33:00.233538000 +0200
@@ -54,7 +54,7 @@
       set_relative_rate(1.0 / resamp_ratio);
       message_port_register_in(pmt::intern("msg_in"));
       set_msg_handler(pmt::intern("msg_in"), boost::bind(
-              &fractional_resampler_cc_impl::handle_msg, this, _1));
+              &fractional_resampler_cc_impl::handle_msg, this, boost::placeholders::_1));
     }
 
     fractional_resampler_cc_impl::~fractional_resampler_cc_impl()
diff -Naur gnuradio-3.7.13.2.orig/gr-filter/lib/fractional_resampler_ff_impl.cc gnuradio-3.7.13.2/gr-filter/lib/fractional_resampler_ff_impl.cc
--- gnuradio-3.7.13.2.orig/gr-filter/lib/fractional_resampler_ff_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-filter/lib/fractional_resampler_ff_impl.cc	2020-05-17 10:33:25.006538000 +0200
@@ -55,7 +55,7 @@
 
       message_port_register_in(pmt::intern("msg_in"));
       set_msg_handler(pmt::intern("msg_in"), boost::bind(
-              &fractional_resampler_ff_impl::handle_msg, this, _1));
+              &fractional_resampler_ff_impl::handle_msg, this, boost::placeholders::_1));
     }
 
     fractional_resampler_ff_impl::~fractional_resampler_ff_impl()
diff -Naur gnuradio-3.7.13.2.orig/gr-filter/lib/freq_xlating_fir_filter_XXX_impl.cc.t gnuradio-3.7.13.2/gr-filter/lib/freq_xlating_fir_filter_XXX_impl.cc.t
--- gnuradio-3.7.13.2.orig/gr-filter/lib/freq_xlating_fir_filter_XXX_impl.cc.t	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-filter/lib/freq_xlating_fir_filter_XXX_impl.cc.t	2020-05-17 10:32:22.116538000 +0200
@@ -69,7 +69,7 @@
       message_port_register_in(pmt::mp("freq"));
       set_msg_handler(pmt::mp("freq"),
                       boost::bind(&@IMPL_NAME@::handle_set_center_freq,
-                                  this, _1));
+                                  this, boost::placeholders::_1));
     }
 
     @IMPL_NAME@::~@IMPL_NAME@()
diff -Naur gnuradio-3.7.13.2.orig/gr-qtgui/lib/const_sink_c_impl.cc gnuradio-3.7.13.2/gr-qtgui/lib/const_sink_c_impl.cc
--- gnuradio-3.7.13.2.orig/gr-qtgui/lib/const_sink_c_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-qtgui/lib/const_sink_c_impl.cc	2020-05-17 10:25:08.165538000 +0200
@@ -69,7 +69,7 @@
       // setup PDU handling input port
       message_port_register_in(pmt::mp("in"));
       set_msg_handler(pmt::mp("in"),
-                      boost::bind(&const_sink_c_impl::handle_pdus, this, _1));
+                      boost::bind(&const_sink_c_impl::handle_pdus, this, boost::placeholders::_1));
 
       for(int i = 0; i < d_nconnections; i++) {
 	d_residbufs_real.push_back((double*)volk_malloc(d_buffer_size*sizeof(double),
diff -Naur gnuradio-3.7.13.2.orig/gr-qtgui/lib/edit_box_msg_impl.cc gnuradio-3.7.13.2/gr-qtgui/lib/edit_box_msg_impl.cc
--- gnuradio-3.7.13.2.orig/gr-qtgui/lib/edit_box_msg_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-qtgui/lib/edit_box_msg_impl.cc	2020-05-17 10:25:08.167538000 +0200
@@ -161,7 +161,7 @@
       message_port_register_in(pmt::mp("val"));
 
       set_msg_handler(pmt::mp("val"),
-                      boost::bind(&edit_box_msg_impl::set_value, this, _1));
+                      boost::bind(&edit_box_msg_impl::set_value, this, boost::placeholders::_1));
     }
 
     edit_box_msg_impl::~edit_box_msg_impl()
diff -Naur gnuradio-3.7.13.2.orig/gr-qtgui/lib/freq_sink_c_impl.cc gnuradio-3.7.13.2/gr-qtgui/lib/freq_sink_c_impl.cc
--- gnuradio-3.7.13.2.orig/gr-qtgui/lib/freq_sink_c_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-qtgui/lib/freq_sink_c_impl.cc	2020-05-17 10:25:08.166538000 +0200
@@ -76,12 +76,12 @@
       message_port_register_out(d_port);
       message_port_register_in(d_port);
       set_msg_handler(d_port,
-                      boost::bind(&freq_sink_c_impl::handle_set_freq, this, _1));
+                      boost::bind(&freq_sink_c_impl::handle_set_freq, this, boost::placeholders::_1));
 
       // setup PDU handling input port
       message_port_register_in(pmt::mp("in"));
       set_msg_handler(pmt::mp("in"),
-                      boost::bind(&freq_sink_c_impl::handle_pdus, this, _1));
+                      boost::bind(&freq_sink_c_impl::handle_pdus, this, boost::placeholders::_1));
 
       d_main_gui = NULL;
 
diff -Naur gnuradio-3.7.13.2.orig/gr-qtgui/lib/freq_sink_f_impl.cc gnuradio-3.7.13.2/gr-qtgui/lib/freq_sink_f_impl.cc
--- gnuradio-3.7.13.2.orig/gr-qtgui/lib/freq_sink_f_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-qtgui/lib/freq_sink_f_impl.cc	2020-05-17 10:25:08.169538000 +0200
@@ -75,12 +75,12 @@
       message_port_register_out(d_port);
       message_port_register_in(d_port);
       set_msg_handler(d_port,
-                      boost::bind(&freq_sink_f_impl::handle_set_freq, this, _1));
+                      boost::bind(&freq_sink_f_impl::handle_set_freq, this, boost::placeholders::_1));
 
       // setup PDU handling input port
       message_port_register_in(pmt::mp("in"));
       set_msg_handler(pmt::mp("in"),
-                      boost::bind(&freq_sink_f_impl::handle_pdus, this, _1));
+                      boost::bind(&freq_sink_f_impl::handle_pdus, this, boost::placeholders::_1));
 
       d_main_gui = NULL;
 
diff -Naur gnuradio-3.7.13.2.orig/gr-qtgui/lib/histogram_sink_f_impl.cc gnuradio-3.7.13.2/gr-qtgui/lib/histogram_sink_f_impl.cc
--- gnuradio-3.7.13.2.orig/gr-qtgui/lib/histogram_sink_f_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-qtgui/lib/histogram_sink_f_impl.cc	2020-05-17 10:25:08.168538000 +0200
@@ -72,7 +72,7 @@
       // setup PDU handling input port
       message_port_register_in(pmt::mp("in"));
       set_msg_handler(pmt::mp("in"),
-                      boost::bind(&histogram_sink_f_impl::handle_pdus, this, _1));
+                      boost::bind(&histogram_sink_f_impl::handle_pdus, this, boost::placeholders::_1));
 
       // +1 for the PDU buffer
       for(int i = 0; i < d_nconnections+1; i++) {
diff -Naur gnuradio-3.7.13.2.orig/gr-qtgui/lib/sink_c_impl.cc gnuradio-3.7.13.2/gr-qtgui/lib/sink_c_impl.cc
--- gnuradio-3.7.13.2.orig/gr-qtgui/lib/sink_c_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-qtgui/lib/sink_c_impl.cc	2020-05-17 10:25:08.166538000 +0200
@@ -79,7 +79,7 @@
       message_port_register_out(d_port);
       message_port_register_in(d_port);
       set_msg_handler(d_port,
-                      boost::bind(&sink_c_impl::handle_set_freq, this, _1));
+                      boost::bind(&sink_c_impl::handle_set_freq, this, boost::placeholders::_1));
 
       d_main_gui = NULL;
 
diff -Naur gnuradio-3.7.13.2.orig/gr-qtgui/lib/sink_f_impl.cc gnuradio-3.7.13.2/gr-qtgui/lib/sink_f_impl.cc
--- gnuradio-3.7.13.2.orig/gr-qtgui/lib/sink_f_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-qtgui/lib/sink_f_impl.cc	2020-05-17 10:25:08.165538000 +0200
@@ -79,7 +79,7 @@
       message_port_register_out(d_port);
       message_port_register_in(d_port);
       set_msg_handler(d_port,
-                      boost::bind(&sink_f_impl::handle_set_freq, this, _1));
+                      boost::bind(&sink_f_impl::handle_set_freq, this, boost::placeholders::_1));
 
       d_main_gui = NULL;
 
diff -Naur gnuradio-3.7.13.2.orig/gr-qtgui/lib/time_raster_sink_b_impl.cc gnuradio-3.7.13.2/gr-qtgui/lib/time_raster_sink_b_impl.cc
--- gnuradio-3.7.13.2.orig/gr-qtgui/lib/time_raster_sink_b_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-qtgui/lib/time_raster_sink_b_impl.cc	2020-05-17 10:25:08.167538000 +0200
@@ -82,7 +82,7 @@
       // setup PDU handling input port
       message_port_register_in(pmt::mp("in"));
       set_msg_handler(pmt::mp("in"),
-                      boost::bind(&time_raster_sink_b_impl::handle_pdus, this, _1));
+                      boost::bind(&time_raster_sink_b_impl::handle_pdus, this, boost::placeholders::_1));
 
       d_scale = 1.0f;
 
diff -Naur gnuradio-3.7.13.2.orig/gr-qtgui/lib/time_raster_sink_f_impl.cc gnuradio-3.7.13.2/gr-qtgui/lib/time_raster_sink_f_impl.cc
--- gnuradio-3.7.13.2.orig/gr-qtgui/lib/time_raster_sink_f_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-qtgui/lib/time_raster_sink_f_impl.cc	2020-05-17 10:25:08.168538000 +0200
@@ -82,7 +82,7 @@
       // setup PDU handling input port
       message_port_register_in(pmt::mp("in"));
       set_msg_handler(pmt::mp("in"),
-                      boost::bind(&time_raster_sink_f_impl::handle_pdus, this, _1));
+                      boost::bind(&time_raster_sink_f_impl::handle_pdus, this, boost::placeholders::_1));
 
       d_icols = static_cast<int>(ceil(d_cols));
       d_tmpflt = (float*)volk_malloc(d_icols*sizeof(float),
diff -Naur gnuradio-3.7.13.2.orig/gr-qtgui/lib/time_sink_c_impl.cc gnuradio-3.7.13.2/gr-qtgui/lib/time_sink_c_impl.cc
--- gnuradio-3.7.13.2.orig/gr-qtgui/lib/time_sink_c_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-qtgui/lib/time_sink_c_impl.cc	2020-05-17 10:25:08.166538000 +0200
@@ -73,7 +73,7 @@
       // setup PDU handling input port
       message_port_register_in(pmt::mp("in"));
       set_msg_handler(pmt::mp("in"),
-                      boost::bind(&time_sink_c_impl::handle_pdus, this, _1));
+                      boost::bind(&time_sink_c_impl::handle_pdus, this, boost::placeholders::_1));
 
       // +2 for the PDU message buffers
       for(int n = 0; n < d_nconnections+2; n++) {
diff -Naur gnuradio-3.7.13.2.orig/gr-qtgui/lib/time_sink_f_impl.cc gnuradio-3.7.13.2/gr-qtgui/lib/time_sink_f_impl.cc
--- gnuradio-3.7.13.2.orig/gr-qtgui/lib/time_sink_f_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-qtgui/lib/time_sink_f_impl.cc	2020-05-17 10:25:08.168538000 +0200
@@ -73,7 +73,7 @@
       // setup PDU handling input port
       message_port_register_in(pmt::mp("in"));
       set_msg_handler(pmt::mp("in"),
-                      boost::bind(&time_sink_f_impl::handle_pdus, this, _1));
+                      boost::bind(&time_sink_f_impl::handle_pdus, this, boost::placeholders::_1));
 
       // +1 for the PDU buffer
       for(int n = 0; n < d_nconnections+1; n++) {
diff -Naur gnuradio-3.7.13.2.orig/gr-qtgui/lib/waterfall_sink_c_impl.cc gnuradio-3.7.13.2/gr-qtgui/lib/waterfall_sink_c_impl.cc
--- gnuradio-3.7.13.2.orig/gr-qtgui/lib/waterfall_sink_c_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-qtgui/lib/waterfall_sink_c_impl.cc	2020-05-17 10:25:08.167538000 +0200
@@ -114,12 +114,12 @@
       message_port_register_out(d_port);
       message_port_register_in(d_port);
       set_msg_handler(d_port,
-                      boost::bind(&waterfall_sink_c_impl::handle_set_freq, this, _1));
+                      boost::bind(&waterfall_sink_c_impl::handle_set_freq, this, boost::placeholders::_1));
 
       // setup PDU handling input port
       message_port_register_in(pmt::mp("in"));
       set_msg_handler(pmt::mp("in"),
-                      boost::bind(&waterfall_sink_c_impl::handle_pdus, this, _1));
+                      boost::bind(&waterfall_sink_c_impl::handle_pdus, this, boost::placeholders::_1));
     }
 
     waterfall_sink_c_impl::~waterfall_sink_c_impl()
diff -Naur gnuradio-3.7.13.2.orig/gr-qtgui/lib/waterfall_sink_f_impl.cc gnuradio-3.7.13.2/gr-qtgui/lib/waterfall_sink_f_impl.cc
--- gnuradio-3.7.13.2.orig/gr-qtgui/lib/waterfall_sink_f_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-qtgui/lib/waterfall_sink_f_impl.cc	2020-05-17 10:25:08.165538000 +0200
@@ -110,12 +110,12 @@
       message_port_register_out(d_port);
       message_port_register_in(d_port);
       set_msg_handler(d_port,
-                      boost::bind(&waterfall_sink_f_impl::handle_set_freq, this, _1));
+                      boost::bind(&waterfall_sink_f_impl::handle_set_freq, this, boost::placeholders::_1));
 
       // setup PDU handling input port
       message_port_register_in(pmt::mp("in"));
       set_msg_handler(pmt::mp("in"),
-                      boost::bind(&waterfall_sink_f_impl::handle_pdus, this, _1));
+                      boost::bind(&waterfall_sink_f_impl::handle_pdus, this, boost::placeholders::_1));
     }
 
     waterfall_sink_f_impl::~waterfall_sink_f_impl()
diff -Naur gnuradio-3.7.13.2.orig/gr-uhd/lib/usrp_block_impl.cc gnuradio-3.7.13.2/gr-uhd/lib/usrp_block_impl.cc
--- gnuradio-3.7.13.2.orig/gr-uhd/lib/usrp_block_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-uhd/lib/usrp_block_impl.cc	2020-05-17 10:25:08.171538000 +0200
@@ -145,11 +145,11 @@
   message_port_register_in(pmt::mp("command"));
   set_msg_handler(
       pmt::mp("command"),
-      boost::bind(&usrp_block_impl::msg_handler_command, this, _1)
+      boost::bind(&usrp_block_impl::msg_handler_command, this, boost::placeholders::_1)
   );
 
 // cuz we lazy:
-#define REGISTER_CMD_HANDLER(key, _handler) register_msg_cmd_handler(key, boost::bind(&usrp_block_impl::_handler, this, _1, _2, _3))
+#define REGISTER_CMD_HANDLER(key, _handler) register_msg_cmd_handler(key, boost::bind(&usrp_block_impl::_handler, this, boost::placeholders::_1, boost::placeholders::_2, boost::placeholders::_3))
   // Register default command handlers:
   REGISTER_CMD_HANDLER(cmd_freq_key(), _cmd_handler_freq);
   REGISTER_CMD_HANDLER(cmd_gain_key(), _cmd_handler_gain);
@@ -266,7 +266,7 @@
     if (not _wait_for_locked_sensor(
             get_mboard_sensor_names(mboard_index),
             sensor_name,
-            boost::bind(&usrp_block_impl::get_mboard_sensor, this, _1, mboard_index)
+            boost::bind(&usrp_block_impl::get_mboard_sensor, this, boost::placeholders::_1, mboard_index)
         )) {
       GR_LOG_WARN(d_logger, boost::format("Sensor '%s' failed to lock within timeout on motherboard %d.") % sensor_name % mboard_index);
       clocks_locked = false;
diff -Naur gnuradio-3.7.13.2.orig/gr-uhd/lib/usrp_source_impl.cc gnuradio-3.7.13.2/gr-uhd/lib/usrp_source_impl.cc
--- gnuradio-3.7.13.2.orig/gr-uhd/lib/usrp_source_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-uhd/lib/usrp_source_impl.cc	2020-05-17 10:25:08.171538000 +0200
@@ -82,7 +82,7 @@
 #ifdef GR_UHD_USE_STREAM_API
       _samps_per_packet = 1;
 #endif
-      register_msg_cmd_handler(cmd_tag_key(), boost::bind(&usrp_source_impl::_cmd_handler_tag, this, _1));
+      register_msg_cmd_handler(cmd_tag_key(), boost::bind(&usrp_source_impl::_cmd_handler_tag, this, boost::placeholders::_1));
     }
 
     usrp_source_impl::~usrp_source_impl()
diff -Naur gnuradio-3.7.13.2.orig/gr-zeromq/lib/pub_msg_sink_impl.cc gnuradio-3.7.13.2/gr-zeromq/lib/pub_msg_sink_impl.cc
--- gnuradio-3.7.13.2.orig/gr-zeromq/lib/pub_msg_sink_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-zeromq/lib/pub_msg_sink_impl.cc	2020-05-17 10:25:08.164538000 +0200
@@ -58,7 +58,7 @@
 
       message_port_register_in(pmt::mp("in"));
       set_msg_handler( pmt::mp("in"),
-        boost::bind(&pub_msg_sink_impl::handler, this, _1));
+        boost::bind(&pub_msg_sink_impl::handler, this, boost::placeholders::_1));
     }
 
     pub_msg_sink_impl::~pub_msg_sink_impl()
diff -Naur gnuradio-3.7.13.2.orig/gr-zeromq/lib/push_msg_sink_impl.cc gnuradio-3.7.13.2/gr-zeromq/lib/push_msg_sink_impl.cc
--- gnuradio-3.7.13.2.orig/gr-zeromq/lib/push_msg_sink_impl.cc	2018-06-13 21:38:34.000000000 +0200
+++ gnuradio-3.7.13.2/gr-zeromq/lib/push_msg_sink_impl.cc	2020-05-17 10:25:08.164538000 +0200
@@ -60,7 +60,7 @@
 
       message_port_register_in(pmt::mp("in"));
       set_msg_handler(pmt::mp("in"),
-                      boost::bind(&push_msg_sink_impl::handler, this, _1));
+                      boost::bind(&push_msg_sink_impl::handler, this, boost::placeholders::_1));
     }
 
     push_msg_sink_impl::~push_msg_sink_impl()
